spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BoxHead.scala

```diff
@@ -283,26 +284,79 @@ private[nn] class BoxPostProcessor(
     }
     val classLogits = input[Tensor[Float]](1)
     val boxRegression = input[Tensor[Float]](2)
-    val bbox = input[Tensor[Float]](3)
+    require(input(3).isInstanceOf[Table], s"bbox should be table, but get ${input(3)}")
+    val bbox = input[Table](3)
+
+    val boxesInImage = new Array[Int](bbox.length())
+    for (i <- 0 to boxesInImage.length - 1) {
+      boxesInImage(i) = bbox[Tensor[Float]](i + 1).size(1)
+    }
 
     if (boxesBuf == null) boxesBuf = Tensor[Float]
     boxesBuf.resizeAs(boxRegression)
+    if (concatBoxes == null) concatBoxes = Tensor[Float]
+    concatBoxes.resize(boxesInImage.sum, 4)
+    var start = 1
+    for (i <- 0 to boxesInImage.length - 1) {
+      val length = boxesInImage(i)
+      concatBoxes.narrow(1, start, length).copy(bbox[Tensor[Float]](i + 1))
+      start += length
+    }
 
     val classProb = softMax.forward(classLogits)
-    BboxUtil.decodeWithWeight(boxRegression, bbox, weight, boxesBuf)
-
-    val boxesInImage = bbox.size(1)
-    val proposalSplit = boxesBuf.split(boxesInImage, dim = 1)
-    val classProbSplit = classProb.split(boxesInImage, dim = 1)
-
-    val roilabels = filterResults(proposalSplit(0), classProbSplit(0), nClasses)
+    BboxUtil.decodeWithWeight(boxRegression, concatBoxes, weight, boxesBuf)
```

Comment:
`BBoxUtil` ?

Reply:
Sorry, it's hard to change as this object has existed for a few release. Users may use in their cases.